src/main/java/com/example/librarymanagementsystem/dao/JournalDAO.java

```diff
@@ -0,0 +1,178 @@
+
+            // Insert into Journal
+            if (resultSet.next()){
+                int resourceId = resultSet.getInt("resourceId");
+
+                preparedStatement = connection.prepareStatement(query1);
+                preparedStatement.setInt(1, resourceId);
+                preparedStatement.setString(2, journal.getAuthor());
+                preparedStatement.setString(3, journal.getGenre());
+                preparedStatement.setInt(4, journal.getVolumeNumber());
+                preparedStatement.setDate(5, journal.getPublicationDate());
+                preparedStatement.setObject(6, "available", Types.OTHER);
+                preparedStatement.executeUpdate();
+            }
+
+            connection.commit();
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    public void updateResource(Journal journal){
+        query = "UPDATE LibraryResource SET title = ?, numberOfCopies = ? WHERE resourceId = ?";
+        query1 = "UPDATE Journal SET author = ?, genre = ?, volumeNumber = ?, publicationDate = ?, statusOfJournalAvailability = ? WHERE resourceId = ?";
+
+        try{
+            connection.setAutoCommit(false);
+
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, journal.getTitle());
+            preparedStatement.setInt(2, journal.getNumberOfCopies());
+            preparedStatement.setInt(3, journal.getResourceId());
+            preparedStatement.executeUpdate();
+
+            preparedStatement = connection.prepareStatement(query1);
+            preparedStatement.setString(1, journal.getAuthor());
+            preparedStatement.setString(2, journal.getGenre());
+            preparedStatement.setInt(3, journal.getVolumeNumber());
+            preparedStatement.setDate(4, journal.getPublicationDate());
+            preparedStatement.setObject(5, "available", Types.OTHER);
+            preparedStatement.setInt(6, journal.getResourceId());
+            preparedStatement.executeUpdate();
+
+            connection.commit();
+        } catch (SQLException e){
+            throw new RuntimeException(e.getMessage());
+        }
+
+    }
+
+    @Override
+    public void deleteResource(int resourceId)  {
+        query = "DELETE FROM Journal WHERE resourceId = ?";
+        query1 = "DELETE FROM LibraryResource WHERE resourceId = ?";
+
+        try{
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setInt(1, resourceId);
+            preparedStatement.executeUpdate();
+
+            preparedStatement = connection.prepareStatement(query1);
+            preparedStatement.setInt(1, resourceId);
+            preparedStatement.executeUpdate();
+
+        } catch (Exception e) {
+            throw new RuntimeException("Journal with Id: " + resourceId +" does not exist." + e.getMessage());
+        }
+    }
```

Comment:
- Introduce rollback to handle transaction failures and remember to commit the changes.
- Handle specific exceptions instead of generic ones